antenna/app.py

```diff
@@ -227,31 +192,31 @@ def extract_payload(self, req):
         raw_crash = {}
         dumps = {}
 
-        for key, val in payload.items():
-            if key == 'dump_checksums':
+        for fs_item in fs.list:
+            if fs_item.name == 'dump_checksums':
                 # We don't want to pick up the dump_checksums from a raw
                 # crash that was re-submitted.
                 continue
 
-            elif isinstance(val, str):
-                raw_crash[key] = de_null(val)
+            elif fs_item.type.startswith('text/plain'):
+                raw_crash[fs_item.name] = de_null(fs_item.value)
 
-            elif isinstance(val, bytes):
+            elif fs_item.type.startswith('application/octet-stream') or isinstance(fs_item.value, bytes):
```

Comment:
Breakpad client in Firefox doesn't send any `text/plain` content-type headers, but it does send `application/octet-stream`. I adjusted the code accordingly.

For example:

```
Content-Disposition: form-data; name="useragent_locale"

en-US
```

and:

```
Content-Disposition: form-data; name="upload_file_minidump"; filename="6da3499e-f6ae-22d6-1e1fdac8-16464a16.dmp"
Content-Type: application/octet-stream

<BYTES>
```